clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/DefaultLaunchConfigurationBuilder.groovy

```diff
@@ -171,10 +178,12 @@ class DefaultLaunchConfigurationBuilder implements LaunchConfigurationBuilder {
 
   }
 
-  private String getUserData(String asgName, String launchConfigName, String region, String account, String environment, String accountType) {
+  private String getUserData(String asgName, String launchConfigName, String region, String account, String environment, String accountType, String base64UserData) {
     String data = userDataProviders?.collect { udp ->
       udp.getUserData(asgName, launchConfigName, region, account, environment, accountType)
     }?.join("\n")
+    String userDataDecoded = new String(base64UserData.decodeBase64())
+    data = [data, userDataDecoded]?.join("\n")
```

Comment:
the `?.` here is not doing what you think. The list will never be null, but the items in it could be. Something like
`[data, userDataDecoded].findResults { it }.join("\n")` would work